projects/gloo/pkg/plugins/grpcweb/plugin.go

```
@@ -52,11 +54,13 @@ func (p *plugin) isDisabled(httplistener *v1.HttpListener) bool {
 	return grpcWeb.GetDisable()
 }
 
+// HttpFilters will add an empty version of the grpcweb filter to a listener
+// if it is not explicitly set.
 func (p *plugin) HttpFilters(params plugins.Params, listener *v1.HttpListener) ([]plugins.StagedHttpFilter, error) {
 	if p.isDisabled(listener) {
 		return nil, nil
 	}
-	return []plugins.StagedHttpFilter{
-		plugins.NewStagedFilter(wellknown.GRPCWeb, pluginStage),
-	}, nil
+	filter, _ := plugins.NewStagedFilterWithConfig(wellknown.GRPCWeb, &envoygrpcweb.GrpcWeb{}, pluginStage)
```

I know we historically would drop the error for nil config, and do that inside of a utility function. Now that its in the plugin, I think a comment outlining why this is safe (or why an error would occur in the first place). Alternatively, we could just return the error as well, even if we are assuming it will be nil almost every time.

---

Yeah so we do return the error always in the WithConfig variant but didnt do it in the version without config.
I updated the comment on NewStagedFilter to reflect the error reasons.

I like the return of the error as the _ shows that we know and are asserting that it should never fail but we dont need to have a different function. If we want to mute the error then we could wrap this with an error swallowing util like
MustNewStagedFilter but in general having a single function to call seems cleaner in general.
That being said I am happy to implement it if there is still a desire.